drivers/SmartThings/philips-hue/src/handlers/refresh_handlers.lua

```diff
@@ -0,0 +1,256 @@
+            conn_rest_err
+          )
+        )
+        return
+      end
+
+      if conn_status.errors and #conn_status.errors > 0 then
+        bridge_device.log.error("Errors in connectivity status payload: " .. st_utils.stringify_table(conn_status.errors))
+        return
+      end
+
+      local conn_status_cache = {}
+
+      for _, zigbee_status in ipairs(conn_status.data) do
+        conn_status_cache[zigbee_status.owner.rid] = zigbee_status
+      end
+
+      local statuses_by_device_type = {}
+      for _, device in ipairs(child_devices) do
+        local device_type = device:get_field(Fields.DEVICE_TYPE)
+        -- Query for the states of all devices for a device type for a given child device,
+        -- but only the first time we encounter a device type. We cache them since we're refreshing
+        -- everything.
+        if
+            type(device_type_refresh_handlers_map[device_type]) == "function" and
+            statuses_by_device_type[device_type] == nil
+        then
+          local reprs, rest_err = hue_api:get_all_reprs_for_rtype(device_type)
+          if reprs ~= nil and rest_err == nil then
+            local cached_states = {}
+            for _, repr_state in ipairs(reprs.data) do
+              cached_states[repr_state.owner.rid] = repr_state
+            end
+            statuses_by_device_type[device_type] = cached_states
+          else
+            log.error("Error refreshing all for resource type [%s]", device_type)
+          end
+        end
+        M.handler_for_device_type(device_type)(
+          driver,
+          device,
+          conn_status_cache,
+          statuses_by_device_type[device_type]
+        )
+      end
+    end,
+    string.format("Refresh All Child Devices for Hue Bridge [%s]",
+      (bridge_device and bridge_device.label) or "Unknown Bridge")
+  )
+end
+
+---@param driver HueDriver
+---@param light_device HueChildDevice
+---@param conn_status_cache table|nil
+---@param light_status_cache table|nil
+function M.do_refresh_light(driver, light_device, conn_status_cache, light_status_cache)
+  local light_resource_id = light_device:get_field(Fields.RESOURCE_ID)
+  local hue_device_id = light_device:get_field(Fields.HUE_DEVICE_ID)
+
+  local do_zigbee_request = true
+  local do_light_request = true
+
+  if type(conn_status_cache) == "table" then
+    local zigbee_status = conn_status_cache[hue_device_id]
+    if zigbee_status ~= nil and zigbee_status.status ~= nil then
+      do_zigbee_request = false
+      if zigbee_status.status == "connected" then
+        light_device.log.debug(string.format("Zigbee Status for %s is connected", light_device.label))
+        light_device:online()
+        light_device:set_field(Fields.IS_ONLINE, true)
+      else
+        light_device.log.debug(string.format("Zigbee Status for %s is not connected", light_device.label))
+        light_device:set_field(Fields.IS_ONLINE, false)
+        light_device:offline()
+      end
+    end
+  end
```

Comment:
Two things:
1. This seems like it could be generalized for all device types? Potentially something that could go in this refactor.
2. Should we be refreshing the state on a device that we know is unreachable by the bridge?

Reply:
@cjswedes I started looking at this yesterday and I realized that if we build this on top of the next two PR's in the stack, we can actually implement a single refresh handler for all devices. So I'll be making this change further along.